src/fungible_token/ft_core.rs

```diff
@@ -0,0 +1,149 @@
+use crate::fungible_token::{io::*, state::*};
+use gstd::{exec, msg, prelude::*, ActorId};
+const ZERO_ID: ActorId = ActorId::new([0u8; 32]);
+
+pub trait FTCore: FTStateKeeper {
+    /// Mints `amount` of token
+    ///
+    /// Requirements: None
+    /// Arguments:
+    /// * `amount`: The amount of token to be minted (actually have no limit)
+    fn mint(&mut self, amount: u128) {
+        self.get_mut()
+            .balances
+            .entry(msg::source())
+            .and_modify(|balance| *balance += amount)
+            .or_insert(amount);
+        self.get_mut().total_supply += amount;
+        msg::reply(
+            FTTransfer {
+                from: ZERO_ID,
+                to: msg::source(),
+                amount,
+            }
+            .encode(),
+            0,
+        )
+        .expect("Error during a replying with FTEvent::FTTransfer");
+    }
+
+    /// Burns `amount` of token
+    ///
+    /// Requirements:
+    /// * msg.sender MUST have enough tokens on his balance
+    /// Arguments:
+    /// `amount`: The amount of token to be burnt
+    fn burn(&mut self, amount: u128) {
+        if self.get().balances.get(&msg::source()).unwrap_or(&0) < &amount {
```

Comment:
May be `expect` is better
```suggestion
        if self.get().balances.get(&msg::source()).expect("The account has no balance at all") < &amount {
```